napari/components/_dims/model.py

```diff
@@ -1,143 +1,255 @@
-from ...util.misc import (compute_max_shape as _compute_max_shape,
-                          guess_metadata)
-from ...util.event import EmitterGroup, Event
-from numpy import clip, integer, ndarray, append, insert, delete
+import numpy as np
 from copy import copy
+from typing import Union, Tuple, Iterable, Sequence
 
-from .view import QtDims
+from ._constants import DimsMode
+from ...util.event import EmitterGroup
 
 
 class Dims:
-    """Dimensions object containing the dimension sliders
+    """Dimensions object modeling multi-dimensional slicing, cropping, and
+    displaying in Napari
 
     Parameters
     ----------
-    viewer : Viewer
-        Parent viewer.
+    init_ndim : int, optional
+        Initial number of dimensions
 
     Attributes
     ----------
-    indices : list
-        Slicing indices controlled by the sliders.
-    max_dims : int
-        Maximum dimensions of the contained layers.
-    max_shape : tuple of int
-        Maximum shape of the contained layers.
+    events : EmitterGroup
+        Event emitter group
+    range : list
+        List of tuples (min, max, step), one for each dimension
```

Comment:
this is what I like to see in parameter descriptions

just don't forget to propagate it up to the type (i.e. `range : list` -> `range : list of 3-tuple`)

Reply:
fixed